core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -464,37 +457,32 @@ public int onStartCommand(Intent intent, int flags, int startId) {
-            Log.e(TAG, "PlaybackService was started with no arguments");
-            stopService();
+            // Typical cases when the service was started with no argument
+            // - when it is first bound, and then moved to startedState, as in <code>serviceManager.moveServiceToStartedState()</code>
+            // - callers (e.g., Controller) explicitly
+            Log.d(TAG, "PlaybackService was started with no arguments.");
             return Service.START_NOT_STICKY;
         }
 
-        if ((flags & Service.START_FLAG_REDELIVERY) != 0) {
-            Log.d(TAG, "onStartCommand is a redelivered intent, calling stopForeground now.");
-            stopForeground(true);
-        } else {
-            if (keycode != -1) {
-                Log.d(TAG, "Received media button event");
-                boolean handled = handleKeycode(keycode, true);
-                if (!handled) {
-                    stopService();
-                    return Service.START_NOT_STICKY;
-                }
-            } else if (!flavorHelper.castDisconnect(castDisconnect) && playable != null) {
-                started = true;
-                boolean stream = intent.getBooleanExtra(EXTRA_SHOULD_STREAM,
-                        true);
-                boolean startWhenPrepared = intent.getBooleanExtra(EXTRA_START_WHEN_PREPARED, false);
-                boolean prepareImmediately = intent.getBooleanExtra(EXTRA_PREPARE_IMMEDIATELY, false);
-                sendNotificationBroadcast(NOTIFICATION_TYPE_RELOAD, 0);
-                //If the user asks to play External Media, the casting session, if on, should end.
-                flavorHelper.castDisconnect(playable instanceof ExternalMedia);
-                if (playable instanceof FeedMedia) {
-                    playable = DBReader.getFeedMedia(((FeedMedia) playable).getId());
-                }
-                mediaPlayer.playMediaObject(playable, stream, startWhenPrepared, prepareImmediately);
+        if (keycode != -1) {
+            Log.d(TAG, "Received media button event");
+            boolean handled = handleKeycode(keycode, true);
+            if (!handled) {
+                serviceManager.stopService(); // TODO: [2716]-RemoveOptional why is it added in the first place in v1.7.0?
+                return Service.START_NOT_STICKY;
+            }
+        } else if (!flavorHelper.castDisconnect(castDisconnect) && playable != null) {
+            boolean stream = intent.getBooleanExtra(EXTRA_SHOULD_STREAM,
+                    true);
+            boolean startWhenPrepared = intent.getBooleanExtra(EXTRA_START_WHEN_PREPARED, false);
+            boolean prepareImmediately = intent.getBooleanExtra(EXTRA_PREPARE_IMMEDIATELY, false);
+            sendNotificationBroadcast(NOTIFICATION_TYPE_RELOAD, 0);
+            //If the user asks to play External Media, the casting session, if on, should end.
+            flavorHelper.castDisconnect(playable instanceof ExternalMedia);
```

Comment:
Can you provide some steps how I can test this on a chromecast?

Reply:
If you're referring to the case `playable` is an `ExternalMedia`, I'm afraid I am not familiar with it.

The main logic here (when service is started with a `Playable`) is largely untouched, compared to v1.7.1 :
https://github.com/AntennaPod/AntennaPod/blob/1.7.1/core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java#L482